### PR TITLE
feat: expose channels state on chat level

### PR DESCRIPTION
**Pull request by MartinCupela**
### 🎯 Goal

Allow the integrators to control the `channels` state outside the `ChannelList` component. Keeping the state inside `ChannelList` forces the SDK maintainers to provide props through which the integrators could somehow impact the channels state, but still not allowing for full control. An example would be that of blocking the integrators from controlling how custom active channel is queried as the default implementation does not allow to customize the query filters etc.

It is unrealistic to keep adding new props, that would hook into different parts of the `ChannelList` code base. Therefore the `channels` state controller is exposed to all the components rendered within the `ChatContext`.

This PR does not introduce breaking changes and therefore the `usePaginatedChannels` hook keeps forwarding the `channels` and `setChannels` values. This should be refactored in the future.

### CODE REVIEW

**Review by oliverlaz — APPROVED** [review]

File: docusaurus/docs/React/components/contexts/chat-context.mdx [hunk 1]
```diff
@@ -99,6 +127,77 @@ You can override the default behavior by pulling it from context and then utiliz
...
+        options={options}
+      />
+      );
+};
+
+```
+
+| Type     |
+| -------- |
+| `Dispatch<SetStateAction<Channel[]>>` |
```
Comment (arnautov-anton) [inline docusaurus/docs/React/components/contexts/chat-context.mdx:199]: Formatting looks a bit messed up here.

File: docusaurus/docs/React/components/contexts/chat-context.mdx [hunk 2]
```diff
@@ -99,6 +127,77 @@ You can override the default behavior by pulling it from context and then utiliz
 | -------- |
 | function |
 
+### setChannels
+
+Sets the list of Channel objects to be rendered by ChannelList component. One have to be careful, when to call `setChannels` as the first channels query executed by the `ChannelList` overrides the whole [`channels` state](#channels). In that case it is better to subscribe to `client` event `channels.queried` and only then set the channels.
```
Comment (arnautov-anton) [inline docusaurus/docs/React/components/contexts/chat-context.mdx:132]:
```suggestion
Sets the list of `Channel` objects to be rendered by `ChannelList` component. One have to be careful, when to call `setChannels` as the first channels query executed by the `ChannelList` overrides the whole [`channels` state](#channels). In that case it is better to subscribe to `client` event `channels.queried` and only then set the channels.
```

File: docusaurus/docs/React/components/contexts/chat-context.mdx [hunk 3]
```diff
@@ -33,6 +33,34 @@ The currently active channel, which populates the [`Channel`](../core-components
...
+| Type        |
+|-------------|
+| `Channel[]` |
+
+### channelsQueryState
+
+Exposes API that:
+
+- indicates, whether and what channels query has been triggered within [`ChannelList` component](../core-components/channel-list.mdx) by its channels pagination controller - `queryInProgress` of type `ChannelQueryState`.
+- allows to set the `queryInProgress` state with `setQueryInProgress` state setter.
```
Comment (arnautov-anton) [inline docusaurus/docs/React/components/contexts/chat-context.mdx:49]:
```suggestion
- allows to set the `queryInProgress` state with `setQueryInProgress` state setter
```

File: docusaurus/docs/React/components/contexts/chat-context.mdx [hunk 4]
```diff
@@ -33,6 +33,34 @@ The currently active channel, which populates the [`Channel`](../core-components
...
+
+| Type        |
+|-------------|
+| `Channel[]` |
+
+### channelsQueryState
+
+Exposes API that:
+
+- indicates, whether and what channels query has been triggered within [`ChannelList` component](../core-components/channel-list.mdx) by its channels pagination controller - `queryInProgress` of type `ChannelQueryState`.
```
Comment (arnautov-anton) [inline docusaurus/docs/React/components/contexts/chat-context.mdx:48]:
```suggestion
- indicates, whether and what channels query has been triggered within [`ChannelList` component](../core-components/channel-list.mdx) by its channels pagination controller - `queryInProgress` of type `ChannelQueryState`
```

File: docusaurus/docs/React/components/contexts/chat-context.mdx (hunk 1 above)
Comment (MartinCupela) [reply docusaurus/docs/React/components/contexts/chat-context.mdx:199]: Will adjust

Comment (MartinCupela) [reply docusaurus/docs/React/components/contexts/chat-context.mdx]: Done

**Review by arnautov-anton — APPROVED** [review]

**Review by oliverlaz — APPROVED** [review]

**Pull request merged**